main.py

```diff
@@ -0,0 +1,69 @@
+from typing import Optional
+
+from fastapi import Depends, Query, FastAPI, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from db.engine import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors_paginated(
+    page: int = Query(ge=0, default=0),
+    size: int = Query(ge=1, le=100),
+    db: Session = Depends(get_db),
+):
+    return crud.get_all_authors_paginated(db, page, size)
+
+
+@app.get("/authors/{author_id}/", response_model=schemas.Author)
+def read_single_author_by_id(author_id: int, db: Session = Depends(get_db)):
+    return crud.get_author_by_id(db, author_id)
```

Comment:
Consider adding error handling for the case where the author with the specified ID does not exist. You can raise an HTTPException with a 404 status code to indicate that the resource was not found.